### PR TITLE
[th/deprecate-lockdown-pt1] deprecate Lockdown API (part 1)

### PR DESCRIPTION
**Pull request by thom311**
No change in behavior, just documentation update and D-Bus annotation (`org.freedesktop.DBus.Deprecated`).

https://github.com/firewalld/firewalld/issues/1266

A future part will remove part of Lockdown code.

Also try to improve how D-Bus methods are deprecated. The ability to annotate methods/signals (with `org.freedesktop.DBus.Deprecated`) really should have been improved in dbus-python project from the start. Maybe I should still send a patch for that. Regardless, even if dbus-python gets the feature today, we couldn't rely on it from firewalld yet.

### CODE REVIEW

**Comment by thom311** [comment]
[Here](https://gitlab.freedesktop.org/dbus/dbus-python/-/merge_requests/30) dbus-python would gain support for an `is_deprecated` argument. If we ever could rely on this, we could drop code from firewalld.

File: doc/xml/firewalld.dbus.xml
Comment (erig0) [inline doc/xml/firewalld.dbus.xml:1]: Please add the CLI help text deprecation notice to `firewall-offline-cmd` as well.

**Review by erig0 — CHANGES_REQUESTED** [review]
This looks great. Add the missing change to `firewall-offline-cmd` and I'll merge.

File: doc/xml/firewalld.dbus.xml
Comment (thom311) [reply doc/xml/firewalld.dbus.xml:1]: done (yes?)

**Review by erig0 — APPROVED** [review]

**Pull request merged**